azbfs/zc_policy_request_log.go

```diff
@@ -111,17 +115,17 @@ func NewRequestLogPolicyFactory(o RequestLogOptions) pipeline.Factory {
 
 func redactSigQueryParam(rawQuery string) (bool, string) {
 	rawQuery = strings.ToLower(rawQuery) // lowercase the string so we can look for ?sig= and &sig=
-	sigFound := strings.Contains(rawQuery, "?sig=")
+	sigFound := strings.Contains(rawQuery, "?"+SigAzure+"=")
 	if !sigFound {
-		sigFound = strings.Contains(rawQuery, "&sig=")
+		sigFound = strings.Contains(rawQuery, "&"+SigAzure+"=")
```

Comment:
Why don't we just search for `sig=`?

This seems like a roundabout way of finding it. I also don't believe "=" is a valid character in a URL outside of the query anyway. Even there, it's escaped if it's in a key (if it even can be in one) or a value.

Reply:
Do you mean why do the two stage search for "?sig=" and then for "&sig="?   If so, that's not something I'm going to change in this PR. It's existing behaviour, and I'm not going to alter it in a PR that is fundamentally unrelated to that that part of the code.  Too much risk of changing behaviour in an unanticipated way.  

(And, now that I think about it, searching for just "sig=" does change behaviour - in theory at least, if not in practice).